### DISCUSSION TITLE
Langfuse observeOpenAI not logging generations using new zodResponseFormat.

### DISCUSSION THREAD

**Discussion by fabstu** [langfuse/langfuse#2858] [Support] [answered]
OpenAI introduced [Structured Outputs](https://openai.com/index/introducing-structured-outputs-in-the-api/). It seems Langfuse does not log calls using it yet. Or am I doing something wrong? See the attached image showing the first completion appearing, but not the second.

```ts
export async function testLangfuse() {
  const trace = langfuse.trace({ name: "testLangfuse" });
  const span = trace.span({ name: "testLangfuseSpan" });
  const llm = observeOpenAI(
    new OpenAI({
      apiKey: process.env["OPENAI_API_KEY"],
    }),
    {
      parent: span,
      generationName: "testLangfuse",
    }
  );
  const chatCompletion = await llm.chat.completions.create({
    messages: [{ role: "system", content: "Say this is a test!" }],
    model: "gpt-3.5-turbo",
    user: "langfuse",
    max_tokens: 300,
  });
  console.log("chatCompletion", chatCompletion);

  // Does not appear in span. zodResponseFormat requires gpt-4o-2024-08-06 or gpt-4o-mini.
  const GeneratedSchema = z.object({
    output: z.string(),
  });
  const chatCompletion2 = await llm.chat.completions.create({
    messages: [{ role: "system", content: "Say this is a test!" }],
    model: "gpt-4o-2024-08-06",
    user: "langfuse",
    max_tokens: 300,
    response_format: zodResponseFormat(GeneratedSchema, "output"),
  });
  console.log("chatCompletion2", chatCompletion2);

  console.log("TraceURL:", trace.getTraceUrl());
  langfuse.flush();
}

```

<img width="1357" alt="Screenshot 2024-08-07 at 23 40 40" src="https://github.com/user-attachments/assets/5468c6ba-2545-43d3-9cb5-6dfaef7fee85">

**Comment by marcklingen** [comment]
thanks for sharing, currently extending support in our python libraries and will look into adding this in js as well

**Reply by fabstu** [reply]
Thank you!

**Comment by marcklingen** [comment]
Issue to track this: https://github.com/langfuse/langfuse/issues/2862

**Comment by hassiebp** [comment, marked as answer]
Hi @fabstu - we have just released a [new JS SDK version](https://github.com/langfuse/langfuse-js/releases/tag/v3.18.0https://github.com/langfuse/langfuse-js/releases/tag/v3.18.0) that parses the `response_format` to the generation's metadata.

Also, please add `await llm.flushAsync()` to the end of your script to also flush the Langfuse client utilized by the OpenAI integration.

**Reply by fabstu** [reply]
Works, thanks man
